Makefile

```diff
@@ -484,6 +484,9 @@ run-tests: all-tests
 test-results: all-tests
 	INCLUDE_SLOW_TESTS=true BINDIR=$(BINDIR) ./test.sh 2>&1 | tee ./test-results
 
+run-examples: all-examples
+		for i in $(EXAMPLES_BINS); do $$i || exit; done
```

Comment:
Any reason why you preferred synthesizing the list of example programs like so, above, rather than inlining the order in the now-deleted function, `test_example_programs()`?

One thing I noticed is that this `for i in ...` will generate the list in some order which may not match the order in which these example programs were sequenced. Upon testing in your branch, I found the order to be:

```
for i in build/release/bin/examples/splinterdb_wide_values_example build/release/bin/examples/splinterdb_intro_example build/release/bin/examples/splinterdb_iterators_example build/release/bin/examples/splinterdb_custom_ipv4_addr_sortcmp_example; do $i || exit; done
```

The `intro` example comes after the `splinterdb_wide_values_example`, which to a newbie user trying to learn the system is non-intuitive.

My thought is this: There are two candidate consumers of these examples: The end-user and the testing-system.

How about we make a small amendment to `Makefile` to support two targets: `test-examples` and `run-examples`?

The former is what you have in the PR right now. This will run all example programs found in the bin dir. And it is sufficient to just call the new modified `test-examples` target as part of nightly CI target (as you have it in the `ci/steps.lib.yml` file).

The latter is end-user oriented. Yes; It requires a manual update to add a new example program to the list. 

But this, then, will also give us a way to **not** have every end-user run some example programs that have an impact. (For instance, I heard both Alex & you remark that an example program that always does a `ping` to an inet-address should not be part of the test `run-examples` target.)

End-user can still run some example program that is omitted from the `run-examples` list to see how they perform; so no functionality is lost.

I tried out a small change to your `Makefile` and the following seems to do the job:

```
@@ -484,9 +484,17 @@ run-tests: all-tests
 test-results: all-tests
        INCLUDE_SLOW_TESTS=true BINDIR=$(BINDIR) ./test.sh 2>&1 | tee ./test-results

-run-examples: all-examples
+test-examples: all-examples
                for i in $(EXAMPLES_BINS); do $$i || exit; done

+# Target for end-user to run example programs in some useful order.
+run-examples: all-examples
+               for i in splinterdb_intro_example \
+               splinterdb_wide_values_example \
+               splinterdb_iterators_example \
+               splinterdb_custom_ipv4_addr_sortcmp_example; \
+         do $(BINDIR)/$(EXAMPLES_DIR)/$$i || exit; done
+
```

Reply:
If there is a specific order you want users to view these examples, how about naming them to make the order clear, e.g.

`splinterdb_exampe_01_intro`
`splinterdb_example_02_wide_values`
etc...

As it is, it is clear only that intro probably comes before the others, but the order in which a beginner should read the others is not at all clear. 

The above naming change would provide guidance both to users that find the examples by browsing the source tree and by users who find them by running `make run-examples`.